src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3076,6 +3102,23 @@ private bool CheckNullAsNonNullableReference(BoundExpression value)
             return true;
         }
 
+        private static bool IsDefaultOfUnconstrainedTypeParameter(BoundExpression expr)
+        {
+            switch (expr.Kind)
+            {
+                case BoundKind.Conversion:
+                    {
+                        var conversion = (BoundConversion)expr;
+                        return conversion.Conversion.Kind == ConversionKind.DefaultOrNullLiteral &&
+                            IsDefaultOfUnconstrainedTypeParameter(conversion.Operand);
```

Comment:
>IsDefaultOfUnconstrainedTypeParameter(conversion.Operand) [](start = 28, length = 57)

Should we also check that the target type is an unconstrained type parameter?

Reply:
Added `// PROTOTYPE` comment.

---
In reply to: [177571093](https://github.com/dotnet/roslyn/pull/25733#discussion_r177571093) [](ancestors = 177571093)